routes/auth.go

```diff
@@ -40,3 +93,287 @@ func AuthMiddleware() gin.HandlerFunc {
 		ctx.Set("authToken", token)
 	}
 }
+
+type LoginRequest struct {
+	Email    string `binding:"required,email"`
+	Password string `binding:"required"`
+}
+
+type tokenResp struct {
+	Token string    `json:"token" example:"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...."`
+	Exp   time.Time `json:"exp"`
+}
+
+type userResp struct {
+	models.UserBase
+	Role string `json:"role" example:"user"`
+}
+
+type organizationResp struct {
+	ID   uuid.UUID `json:"id" example:"d45ddfc0-1fc3-4fa7-95e9-edd7d134103a"`
+	Role string    `json:"role" example:"user"`
+}
+
+type LoginResponse struct {
+	User         userResp         `json:"user"`
+	Organization organizationResp `json:"organization"`
+	RefreshToken tokenResp        `json:"refreshToken"`
+	AccessToken  tokenResp        `json:"accessToken"`
+}
+
+func getGlobalRole(user *models.User) string {
+	for _, role := range user.GlobalRoles {
+		if role.Role == models.Admin {
+			return "admin"
+		}
+	}
+	return "user"
+}
+
+// Login godoc
+// @Summary    log in using email and password - get refresh tokenResp and initial access tokenResp
+// @Tags       auth
+// @Produce    json
+// @Param      credentials                     body        LoginRequest      true    "User Credentials"
+// @Success    200                             {object}    LoginResponse
+// @Failure    400                             {object}    HTTPErrorResponse
+// @Failure    401                             {object}    HTTPErrorResponse
+// @Failure    500                             {object}    HTTPErrorResponse
+// @Router     /auth/login                     [post]
+func Login(ctx *gin.Context) {
+	log, logCtx := logging.GetRequestLogger(ctx)
+
+	body := LoginRequest{}
+	if err := ctx.ShouldBindJSON(&body); err != nil {
+		log.Warn().Err(err).Msg("validation failed")
+		SendError(ctx, http.StatusBadRequest, err)
+		return
+	}
+
+	user := models.User{
+		UserBase: models.UserBase{
+			Email: body.Email,
+		},
+	}
+
+	db := models.GetDB(logCtx)
+	tx := db.
+		Preload("Organizations").
+		Preload("GlobalRoles").
+		Where(&user).
+		First(&user)
+
+	if err := tx.Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			// we return the same error message in case the email was not found and in case the password was wrong
+			// thus an unauthorized third party is not able to check whether the email is registered with us or not
+			SendError(ctx, http.StatusUnauthorized, errors.New("email or password invalid"))
+		} else {
+			HandleDBError(ctx, logCtx, err)
+		}
+		return
+	}
+
+	if err := bcrypt.CompareHashAndPassword([]byte(user.PwBcrypt), []byte(body.Password)); err != nil {
+		log.
+			Info().
+			Err(err).
+			Str("user_id", user.ID.String()).
+			Msg("Invalid Password")
+		SendError(ctx, http.StatusUnauthorized, errors.New("email or password invalid"))
+		return
+	}
+
+	if len(user.Organizations) == 0 {
+		log.
+			Warn().
+			Str("user_id", user.ID.String()).
+			Msg("no orgs")
```

Comment:
WHAT IS THAT?!

Reply:
Very insightful, I assume you don't like the message?

Reply:
I dislike the indentation and dots at the end of line 🗡️